File: game.py
class Heroi:

    def __init__(self,nome,idade,tipo):
        self.nome=nome
        self.idade=idade
        self.tipo=tipo
        self.tipoAtaque="arma não definida"

    def heroiAtacar(self):

        if self.tipo=="mago":
            self.tipoAtaque="magia"
        elif self.tipo=="guerreiro":
            self.tipoAtaque="espada"
        elif self.tipo=="monge":
            self.tipoAtaque="artes marciais"
        elif self.tipo=="ninja":
            self.tipoAtaque="shuriken"

        print(f"O heroi {self.nome} do tipo {self.tipo} atacou usando {self.tipoAtaque}")


herois=[
Heroi("Goku",20,"monge"),
Heroi("Naruto",15,"ninja"),
Heroi("Samurai Shadow",33,"guerreiro")
]

for h in herois:
    h.heroiAtacar()

File: game.go
package main

import "fmt"

type Heroi struct {
	nome string
	idade int
	tipo string
}

func (h Heroi) heroiAtacar(){

	tipoAtaque := "arma não definida"

	if h.tipo=="mago"{
		tipoAtaque="magia"
	}else if h.tipo=="guerreiro"{
		tipoAtaque="espada"
	}else if h.tipo=="monge"{
		tipoAtaque="artes marciais"
	}else if h.tipo=="ninja"{
		tipoAtaque="shuriken"
	}

	fmt.Printf("O heroi %s do tipo %s atacou usando %s\n",h.nome,h.tipo,tipoAtaque)
}

func main(){

	herois := []Heroi{
		{"Goku",20,"monge"},
		{"Naruto",15,"ninja"},
		{"Samurai Shadow",33,"guerreiro"},
	}

	for _,h := range herois{
		h.heroiAtacar()
	}
}
